tools/flagd-http-connector/src/main/java/dev/openfeature/contrib/tools/flagd/resolver/process/storage/connector/sync/http/HttpConnectorOptions.java

```diff
@@ -0,0 +1,173 @@
+    @Builder.Default
+    private Boolean useFailsafeCache;
+    @Builder.Default
+    private Boolean usePollingCache;
+    @NonNull
+    private String url;
+
+    /**
+     * HttpConnectorOptions constructor.
+     */
+    @Builder
+    public HttpConnectorOptions(Integer pollIntervalSeconds, Integer linkedBlockingQueueCapacity,
+            Integer scheduledThreadPoolSize, Integer requestTimeoutSeconds, Integer connectTimeoutSeconds, String url,
+            Map<String, String> headers, ExecutorService httpClientExecutor, String proxyHost, Integer proxyPort,
+            PayloadCacheOptions payloadCacheOptions, PayloadCache payloadCache, Boolean useHttpCache,
+            Boolean useFailsafeCache, Boolean usePollingCache) {
+        validate(url, pollIntervalSeconds, linkedBlockingQueueCapacity, scheduledThreadPoolSize, requestTimeoutSeconds,
+            connectTimeoutSeconds, proxyHost, proxyPort, payloadCacheOptions, payloadCache, useFailsafeCache,
+            usePollingCache);
+        if (pollIntervalSeconds != null) {
+            this.pollIntervalSeconds = pollIntervalSeconds;
+        }
+        if (linkedBlockingQueueCapacity != null) {
+            this.linkedBlockingQueueCapacity = linkedBlockingQueueCapacity;
+        }
+        if (scheduledThreadPoolSize != null) {
+            this.scheduledThreadPoolSize = scheduledThreadPoolSize;
+        }
+        if (requestTimeoutSeconds != null) {
+            this.requestTimeoutSeconds = requestTimeoutSeconds;
+        }
+        if (connectTimeoutSeconds != null) {
+            this.connectTimeoutSeconds = connectTimeoutSeconds;
+        }
+        this.url = url;
+        if (headers != null) {
+            this.headers = headers;
+        }
+        if (httpClientExecutor != null) {
+            this.httpClientExecutor = httpClientExecutor;
+        }
+        if (proxyHost != null) {
+            this.proxyHost = proxyHost;
+        }
+        if (proxyPort != null) {
+            this.proxyPort = proxyPort;
+        }
+        if (payloadCache != null) {
+            this.payloadCache = payloadCache;
+        }
+        if (payloadCacheOptions != null) {
+            this.payloadCacheOptions = payloadCacheOptions;
+        }
+        if (useHttpCache != null) {
+            this.useHttpCache = useHttpCache;
+        }
+        if (useFailsafeCache != null) {
+            this.useFailsafeCache = useFailsafeCache;
+        }
+        if (usePollingCache != null) {
+            this.usePollingCache = usePollingCache;
+        }
+    }
+
+    @SneakyThrows
+    private void validate(String url, Integer pollIntervalSeconds, Integer linkedBlockingQueueCapacity,
+            Integer scheduledThreadPoolSize, Integer requestTimeoutSeconds, Integer connectTimeoutSeconds,
+            String proxyHost, Integer proxyPort, PayloadCacheOptions payloadCacheOptions,
+            PayloadCache payloadCache, Boolean useFailsafeCache, Boolean usePollingCache) {
+        new URL(url).toURI();
```

Comment:
[nit] this line will be most likely refactored out by the next person, as it seems like a left over, although it is a validation. We should mark this with a comment, or better handle this validation step

Reply:
updated